File: evidence/generated-systems/golang/governance-on-golang/governance-on-golang/internal/model/AuditEngagement.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// AuditEngagement Declaration
//==============================================================
type AuditEngagement struct {
    gorm.Model
     Title                                    string
    StartDate                                                            time.Time
    EndDate                                                            time.Time
    AuditProgramId         *uint
    AuditProgram           *AuditProgram `gorm:"foreignKey:AuditProgramId"`
     BusinessUnits           []BusinessUnit `gorm:"foreignKey:BusinessUnitsFromAuditEngagementId"`
     ControlTests           []ControlTest_ `gorm:"foreignKey:ControlTestsFromAuditEngagementId"`
     Workpapers           []AuditWorkpaper `gorm:"foreignKey:WorkpapersFromAuditEngagementId"`
     Findings           []AuditFinding `gorm:"foreignKey:FindingsFromAuditEngagementId"`
    Status                      AuditStatus

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/governance-on-rails/governanceonrails/app/models/auditengagement.rb
class AuditEngagement < ApplicationRecord
  enum Status: [:Planned, :Fieldwork, :Reporting, :Closed, :OnHold]


  has_many :AuditProgram, class_name: 'AuditProgram'
  has_many :BusinessUnits, class_name: 'BusinessUnit'
  has_many :ControlTests, class_name: 'ControlTest_'
  has_many :Workpapers, class_name: 'AuditWorkpaper'
  has_many :Findings, class_name: 'AuditFinding'

end
